src/components/Editor/QuillEditor.jsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useRef, useState } from "react";
+import "../../css/quillEditor.css";
+import { useFirebase } from "react-redux-firebase";
+import { useDispatch, useSelector } from "react-redux";
+import { Prompt } from "react-router-dom";
+import { setCurrentStep } from "../../store/actions";
+import * as Y from "yjs";
+import { QuillBinding } from "y-quill";
+import Quill from "quill";
+import QuillCursors from "quill-cursors";
+import { FirestoreProvider, getColor } from "@gmcfall/yjs-firestore-provider";
+
+import { onlineFirebaseApp } from "../../config";
+
+Quill.register("modules/cursors", QuillCursors);
+
+const QuillEditor = ({ id, data, tutorial_id }) => {
+  const [allSaved, setAllSaved] = useState(true);
+  const editorRef = useRef(null);
+  const containerRef = useRef(null);
+  let noteID = id || "test_note";
+  const firebase = useFirebase();
+  const dispatch = useDispatch();
+  const basePath = ["cl_codelabz", "organization", "codelabzorg", tutorial_id];
+  var provider, binding;
+  var ydoc;
```

Comment:
If you are declaring three variables, can this be done on the same line together? like `var provider, binding, ydoc` ? What's your thought?

Reply:
yep, we can. I'll change that.

Reply:
Sounds cool!